lib/tapioca/rbi/model.rb

```diff
@@ -0,0 +1,405 @@
+        super()
+        @name = name
+      end
+    end
+
+    class Class < Scope
+      extend T::Sig
+
+      sig { returns(String) }
+      attr_accessor :name
+
+      sig { returns(T.nilable(String)) }
+      attr_accessor :superclass_name
+
+      sig { params(name: String, superclass_name: T.nilable(String)).void }
+      def initialize(name, superclass_name: nil)
+        super()
+        @name = name
+        @superclass_name = superclass_name
+      end
+    end
+
+    class SingletonClass < Scope
+      extend T::Sig
+
+      sig { void }
+      def initialize
+        super()
+      end
+    end
+
+    # Consts
+
+    class Const < Node
+      extend T::Sig
+
+      sig { returns(String) }
+      attr_reader :name, :value
+
+      sig { params(name: String, value: String).void }
+      def initialize(name, value)
+        super()
+        @name = name
+        @value = value
+      end
+    end
+
+    # Methods and args
+
+    class Method < Node
+      extend T::Sig
+
+      sig { returns(String) }
+      attr_accessor :name
+
+      sig { returns(T::Array[Param]) }
+      attr_reader :params
+
+      sig { returns(T::Boolean) }
+      attr_accessor :is_singleton
+
+      sig { returns(Visibility) }
+      attr_accessor :visibility
+
+      sig { returns(T::Array[Sig]) }
+      attr_accessor :sigs
+
+      sig do
+        params(
+          name: String,
+          params: T::Array[Param],
+          is_singleton: T::Boolean,
+          visibility: Visibility,
+          sigs: T::Array[Sig]
+        ).void
+      end
+      def initialize(name, params: [], is_singleton: false, visibility: Visibility::Public, sigs: [])
+        super()
+        @name = name
+        @params = params
+        @is_singleton = is_singleton
+        @visibility = visibility
+        @sigs = sigs
+      end
+
+      sig { params(param: Param).void }
+      def <<(param)
+        @params << param
+      end
+    end
+
+    class Param < Node
+      extend T::Sig
+
+      sig { returns(String) }
+      attr_reader :name
+
+      sig { params(name: String).void }
+      def initialize(name)
+        super()
+        @name = name
+      end
+    end
+
+    class OptParam < Param
+      extend T::Sig
+
+      sig { returns(String) }
+      attr_reader :value
+
+      sig { params(name: String, value: String).void }
+      def initialize(name, value)
+        super(name)
+        @value = value
+      end
+    end
+
+    class RestParam < Param; end
+    class KwParam < Param; end
+    class KwOptParam < OptParam; end
+    class KwRestParam < Param; end
+    class BlockParam < Param; end
+
+    # Mixins
+
+    class Mixin < Node
+      extend T::Sig
+      extend T::Helpers
+
+      abstract!
+
+      sig { returns(String) }
+      attr_reader :name
+
+      sig { params(name: String).void }
+      def initialize(name)
+        super()
+        @name = name
+      end
+    end
+
+    class Include < Mixin; end
+    class Extend < Mixin; end
```

Comment:
Do we also need `prepend`?

Reply:
Sorbet static doesn't support it so we generate `include` instead: https://github.com/Shopify/tapioca/blob/master/lib/tapioca/compilers/symbol_table/symbol_generator.rb#L401-L404

Reply:
I didn't add it as to avoid people trying to use it and not getting the expected behaviour.